### PR TITLE
Refactor flow tests

### PR DESCRIPTION
**Pull request by lightster**
Break tests into pieces so code can be re-used within the test suite.

Another test will be added to the flow test suite when job merging is
added, so I want to refactor this test before we add more duplication
to the test suite.

### CODE REVIEW

**Comment by lightster** [comment]
Hi @Twenty7,

Can you please give this a quick look? It's limited to one file, but at first glance it looks like a lot of changes. None of the changes should result in a functional change.

Thanks,

Lightster

File: tests/src/Hodor/FlowTest.php [hunk 1]
```diff
@@ -52,69 +50,117 @@ public function testJobCanBeRan()
...
+    }
 
-        $this->runCommand("php {$bin_path}/superqueuer.php -c {$this->e_config_file}");
+    /**
+     * @param string $bin
+     * @return string
+     */
+    private function getBinPath($bin)
+    {
+        return escapeshellarg(__DIR__ . '/../../../bin/' . $bin);
```
Comment (Twenty7) [inline tests/src/Hodor/FlowTest.php:81]: I'm not sure escapeshellarg and escapeshellcmd escape what they are supposed to properly on an entire command or not.

**Comment by Twenty7** [comment]
Aside from my comment on escapeshellargs, everything looks good.

**Comment by lightster** [comment]
Hi @Twenty7,

I played around with running `escapeshellarg()` on the path to the command and it seems to work as expected. It is being in a limited case right now—just in those few tests—so it should be fine. We can adjust it later if necessary.

Thanks so much for all of the reviews!

Lightster

**Pull request merged**